guides/2021-05-02-build-flags-backwards-compatibility/en.markdown

```diff
@@ -0,0 +1,163 @@
+---
+layout: post
+title:  "Using build flags for backwards compatibility"
+excerpt: "How to leverage go build flags to avoid breaking downstream projects "
+difficulty: Intermediate
+category: Next steps
+---
+
+_By [Marcos Nils](https://twitter.com/marcosnils), [Docker Captain](https://www.docker.com/captains/marcos-lilljedahl), Hashicorp Ambassador, Go developer, and co-creator of `play-with-go.dev`._
+
+Sometimes, when a new `go` version is released, it also ships with a bunch of changes and really interesting features on the standard library.
+As the time of this article, [go 1.16](https://golang.org/doc/go1.16) has been released around two months ago which introduces some changes and
+new features into the [core library](https://golang.org/doc/go1.16#library) like the new `io.FS` interface, the `go:embed` directive amongst others.
+
+As a module author, how could I introduce these new features and at the same time provide some guarantees that
+my module can still support the last N releases of go?
+
+This guide explains how to deal with situations where you want to use new features of recent versions of `go`
+and at the same time you don't want to force downstream dependecies to upgrade. In this case we'll be using
```

Comment:
".. and at the same time not force downstream users to upgrade"?

"Dependencies" is, I think, the wrong word here, unless it's regularly used like this? Because a dependency is generally something required by your module, no?

"Downstream users" or "downstream dependants" perhaps?

Indeed you use "clients" later which I very much like too.